moduleroot/Gemfile

```diff
@@ -110,79 +95,15 @@ gem 'puppet', *location_for(ENV['PUPPET_GEM_VERSION'])
-    gem "facter",">= 1.6.11","<= 1.7.5",:require => false unless ENV['FACTER_GEM_VERSION']
-    # If hieraversion hasn't been specified and we are
-    # looking for a Puppet Gem version less than 3.5.0, we
-    # need to ensure we get a good Hiera for specs.
-    gem "hiera",">= 1.0.0","<= 1.3.0",:require => false unless ENV['HIERA_GEM_VERSION']
-  end
-end
-
-if explicitly_require_windows_gems
-  # This also means Puppet Gem less than 3.5.0 - this has been tested back
-  # to 3.0.0. Any further back is likely not supported.
-  if puppet_gem_location == :gem
-    gem "ffi", "1.9.0",                          :require => false
-    gem "win32-eventlog", "0.5.3","<= 0.6.5",    :require => false
-    gem "win32-process", "0.6.5","<= 0.7.5",     :require => false
-    gem "win32-security", "~> 0.1.2","<= 0.2.5", :require => false
-    gem "win32-service", "0.7.2","<= 0.8.8",     :require => false
-    gem "minitar", "0.5.4",                      :require => false
-  else
-    gem "ffi", "~> 1.9.0",                       :require => false
-    gem "win32-eventlog", "~> 0.5","<= 0.6.5",   :require => false
-    gem "win32-process", "~> 0.6","<= 0.7.5",    :require => false
-    gem "win32-security", "~> 0.1","<= 0.2.5",   :require => false
-    gem "win32-service", "~> 0.7","<= 0.8.8",    :require => false
-    gem "minitar", "~> 0.5.4",                   :require => false
-  end
-
-  gem "win32-dir", "~> 0.3","<= 0.4.9", :require => false
-  gem "win32console", "1.3.2",          :require => false if RUBY_VERSION =~ /^1\./
-
-  # sys-admin was removed in Puppet 3.7.0+, and doesn't compile
-  # under Ruby 2.3 - so restrict it to Ruby 1.x
-  gem "sys-admin", "1.5.6",             :require => false if RUBY_VERSION =~ /^1\./
-
-  # Puppet less than 3.7.0 requires these.
-  # Puppet 3.5.0+ will control the actual requirements.
-  # These are listed in formats that work with all versions of
-  # Puppet from 3.0.0 to 3.6.x. After that, these were no longer used.
-  # We do not want to allow newer versions than what came out after
-  # 3.6.x to be used as they constitute some risk in breaking older
-  # functionality. So we set these to exact versions.
-  gem "win32-api", "1.4.8",             :require => false
-  gem "win32-taskscheduler", "0.2.2",   :require => false
-  gem "windows-api", "0.4.3",           :require => false
-  gem "windows-pr",  "1.2.3",           :require => false
-else
-  if Gem::Platform.local.os == 'mingw32'
-    # If we're using a Puppet gem on windows, which handles its own win32-xxx gem dependencies (Pup 3.5.0 and above), set maximum versions
-    # Required due to PUP-6445
-    gem "win32-dir", "<= 0.4.9",        :require => false
-    gem "win32-eventlog", "<= 0.6.5",   :require => false
-    gem "win32-process", "<= 0.7.5",    :require => false
-    gem "win32-security", "<= 0.2.5",   :require => false
-    gem "win32-service", "<= 0.8.8",    :require => false
-  end
+if Gem::Platform.local.os == 'mingw32'
```

Comment:
So @jpogran and I went through a number of scenarios for `PUPPET_GEM_VERSION`, and discovered that this does not quite preserve the prior behavior.

What we'd really want is the following to preserve the other condition I was referring to for `explicitly_require_windows_gems`.

That said, we further discovered that these dependencies are no longer necessary to explicitly list for the `file` uri or `https://` git style uris any longer because of the inclusion of the `puppet-module-win-default-r*` gems which have the same dependencies.

Therefore all of this can actually be removed due to that.


```ruby
puppet_gem_location = gem_type(ENV['PUPPET_GEM_VERSION'])
if puppet_gem_location != :gem && Gem::Platform.local.os == 'mingw32'
```

Reply:
Addendum: I think the removal is OK as we shouldn't care about older versions (4.5.2 and below where we would have to at least have `win32-service <= 0.8.8` due to [PUP-6445](https://tickets.puppetlabs.com/browse/PUP-6445))

Reply:
We have control over the `puppet-module-win-default-r*` gems, so we could add the restriction there, if that helps?

Reply:
@DavidS I'm not sure which is better / worse for the `win32-service` restriction. Having in `Gemfile` here is easier to change, but at the same time I don't see us ever upgrading past `0.8.8` so if its set in the ubergem it's easy for projects to get that limitation. Ultimately we want to remove that gem as a dep anyhow.

I think for now we can just update this PR to remove the whole `if Gem::Platform.local.os == 'mingw32'` part.  I don't think we care about Puppet 4.5.2 or lower given LTS is 4.10.

Reply:
Ok, will do!